### ISSUE TITLE
RxCocoa: UIImagePickerController extension causing iTunes Connect to require NSPhotoLibraryUsageDescription

### ISSUE THREAD

**Issue by nighthawk** [ReactiveX/RxSwift#874]
**Short description of the issue**:

The `UIImagePickerController` in RxCocoa causes iTunes Connect to complain about missing `NSPhotoLibraryUsageDescription` key in the `Info.plist` file.

**Self contained code example that reproduces the issue**:

```swift
import RxCocoa
```

**Xcode version**:

```
Xcode 8 GM
```

**Expected outcome**:

Can upload binary to iTunes Connect without issues.

**What actually happens**:

iTunes Store complains about an invalid binary:

> We have discovered one or more issues with your recent delivery for `\(appName)`. To process your delivery, the following issues must be corrected:
> This app attempts to access privacy-sensitive data without a usage description. The app's Info.plist must contain an NSPhotoLibraryUsageDescription key with a string value explaining to the user how the app uses this data.
> Once these issues have been corrected, you can then redeliver the corrected binary.

I have confirmed that RxCocoa is the issue by deleting the extension and uploading again—this time it went through without issue.

The proposal is to remove the `UIImagePickerController` extensions from RxCocoa and, ideally, to move them to a repo in the [RxSwift Community](https://github.com/RxSwiftCommunity).

Related links:

- [Apple Developer forum discussion builds getting rejected](https://forums.developer.apple.com/message/176657#176657)
- [WWDC 2016 talk about usage description keys](https://developer.apple.com/videos/play/wwdc2016/709/?time=1754)

**Installation method**:
  - [x] CocoaPods
  - [ ] Carthage
  - [ ] Git submodules

**Comment by kzaher** [comment]
Hi @nighthawk ,

Ok, I guess we can remove this from RxCocoa. 

I think we also need to move CLLocation manager extensions. Has anyone had issues because of `CLLocationManager` extensions?

**Comment by nighthawk** [comment]
I've just uploaded a build for an app that does not use `CLLocationManager` but includes RxCocoa. iTC only complained about `NSPhotoLibraryUsageDescription` missing. So seems like the `CLLocationManager` extensions are safe to keep—at least for now.

**Comment by mohsenr** [comment]
I can confirm that I had the same issue, and removing `RxImagePickerDelegateProxy.swift` from the target fixed it. So @kzaher’s commit above should work.

**Comment by kzaher** [comment]
This should be fixed now, please reopen the issue if not.

**Comment by egosapien** [comment]
I just integrated RxCocoa and RxSwift into a project (to add better behaviour on search input). The changes were relatively minor, but upon submitting a built to TestFlight it was rejected because "NSPhotoLibraryUsageDescription" was missing from the Info.plist. 

I tried submitting a build where I excluded UIImagePickerController+rx.swift and RxImagePickerDelegateProxy.swift from the target and this was accepted by TestFlight.

Basically, I think this issue is still an issue. A potential solution would be to add some more macros to allow users to exclude these unnecessary files in their builds?
